### PR TITLE
JAMES-2456 Upgrade Tika

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

File: mailbox/tika/src/main/java/org/apache/james/mailbox/tika/TikaHttpClientImpl.java [hunk 1]
```diff
@@ -57,6 +60,7 @@ public InputStream recursiveMetaDataAsJson(InputStream inputStream, String conte
                     .returnContent()
                     .asStream();
         } catch (IOException e) {
+            LOGGER.error("Failing to call Tika", e);
```
Comment (rouazana) [inline mailbox/tika/src/main/java/org/apache/james/mailbox/tika/TikaHttpClientImpl.java]: why do you need to log here if you throw? It should be logged by the caller, no?

File: mailbox/tika/src/main/java/org/apache/james/mailbox/tika/TikaTextExtractor.java [hunk 2]
```diff
@@ -119,6 +123,10 @@ public ContentAndMetadata deserialize(JsonParser jsonParser, DeserializationCont
         private static final String TIKA_HEADER = "X-TIKA";
         private static final String CONTENT_METADATA_HEADER_NAME = TIKA_HEADER + ":content";
 
+        public static ContentAndMetadata empty() {
+            return new ContentAndMetadata(null, ImmutableMap.of());
```
Comment (rouazana) [inline mailbox/tika/src/main/java/org/apache/james/mailbox/tika/TikaTextExtractor.java]: using a private constructor to put a (non named) null value seems bad.

File: mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TikaTextExtractorTest.java [hunk 3]
```diff
@@ -183,7 +188,9 @@ public void deserializerShouldThrowWhenNodeIsNotAnObject() throws Exception {
         expectedException.expectMessage("The element should be a Json object");
 
         TikaTextExtractor textExtractor = new TikaTextExtractor(
-            new NoopMetricFactory(), (inputStream, contentType) -> new ByteArrayInputStream("[\"value1\"]".getBytes(StandardCharsets.UTF_8)));
+            new NoopMetricFactory(), 
+            (inputStream, contentType) -> Optional.of(new ByteArrayInputStream("[\"value1\"]"
+                                                        .getBytes(StandardCharsets.UTF_8))));
 
         InputStream inputStream = null;
         textExtractor.extractContent(inputStream, "text/plain");
```
Comment (rouazana) [inline mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TikaTextExtractorTest.java]: no new test regarding empty behavior?

File: mailbox/tika/src/main/java/org/apache/james/mailbox/tika/TikaHttpClientImpl.java (hunk 1 above)
Comment (rouazana) [reply mailbox/tika/src/main/java/org/apache/james/mailbox/tika/TikaHttpClientImpl.java]: ok fixed later

File: mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TikaTextExtractorTest.java (hunk 3 above)
Comment (aduprat) [reply mailbox/tika/src/test/java/org/apache/james/mailbox/tika/TikaTextExtractorTest.java]: There's already one: `TikaTextExtractorTest.textualContentShouldReturnNullWhenInputStreamIsEmpty()`

File: mailbox/tika/src/main/java/org/apache/james/mailbox/tika/TikaTextExtractor.java [hunk 4]
```diff
@@ -150,6 +150,10 @@ private static String content(Map<String, List<String>> contentAndMetadataMap) {
         private final String content;
         private final Map<String, List<String>> metadata;
 
+        private ContentAndMetadata() {
+            this(null, ImmutableMap.of());
```
Comment (rouazana) [inline mailbox/tika/src/main/java/org/apache/james/mailbox/tika/TikaTextExtractor.java]: then getContent() will return null. We tend to not return null in our classes.

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
```
Tests in error: 
  AccessTokenRepositoryTest.validTokenMustBeRetrieved » Completion org.apache.ja...
```
test this please

**Comment by aduprat** [comment]
```
Tests in error: 
  SpamAssassinContract.spamShouldBeDeliveredInSpamMailboxOrInboxWhenMultipleRecipientsConfigurations:575 » ConditionTimeout
```
test this please

**Comment by chibenwa** [comment]
merged

**Pull request closed without merging**